src/index.js

```diff
@@ -1,7 +1,48 @@
 import * as React from 'react';
 
-import createComponents from './createComponents';
+const Context = React.createContext();
 
-export const { createTimeProvider, TimeProvider, GetTime, withTime } = createComponents(
-  React.createContext(),
+// We return the ISO string version of the date. This is nice because it is not
+// a mutable object like `Date` and it is also a human readable representation.
+const defaultGetTime = (/* props */) => new Date().toISOString();
+
+const cleanProps = ({ interval, children, ...props }) => props;
+
+// We export a TimeProvider factory if they need to override how we `getTime`.
+export const createTimeProvider = getTime => ({ interval = 500, children, ...props }) => {
+  const [currentTime, setCurrentTime] = React.useState(getTime(props));
+
+  React.useEffect(
+    () => {
+      const intervalId = setInterval(() => {
+        setCurrentTime(getTime(props));
+      }, interval);
+
+      return () => {
+        clearInterval(intervalId);
+      };
+    },
+    [setCurrentTime],
+  );
+
+  return <Context.Provider value={{ currentTime, ...props }}>{children}</Context.Provider>;
+};
+
+export const TimeProvider = createTimeProvider(defaultGetTime);
+
+export const GetTime = Context.Consumer;
+
+export const withTime = WrappedComponent => props => (
+  <GetTime>
+    {({ currentTime }) => (
```

Comment:
It's possible to give a `TimeProvider` more props, but they get thrown out here. Should a `TimeProvider` not take more props, or should the extra props also get forwarded to `WrappedComponent`?

Reply:
I think you are right that they should be forwarded